website/static/js/file-browser.js

```diff
@@ -0,0 +1,290 @@
+    this.url = url;
+    this.type = 'breadcrumb' || type;
+};
+
+var Collection = function(label, path, pathQuery) {
+    this.type = 'collection';
+    this.label = label || 'New Collection';
+    this.path = path;
+    this.pathQuery = pathQuery;
+};
+
+var Filter = function (label, data, type) {
+    this.label = label;
+    this.data = data;
+    this.type = 'name' || type;
+};
+
+/**
+ * Initialize File Browser. Prepeares an option object within FileBrowser
+ * @constructor
+ */
+var FileBrowser = {
+    controller : function (options) {
+        var self = this;
+        self.isLoadedUrl = false;
+        self.wrapperSelector = options.wrapperSelector;
+
+        // VIEW STATES
+        self.showInfo = m.prop(false);
+
+        // DEFAULT DATA -- to be switched with server response
+        self.collections = [
+            new Collection('All My Projects', 'users/me/nodes/', { 'related_counts' : true }),
+            new Collection('All My Registrations', 'registrations/', { 'related_counts' : true }),
+            new Collection('Everything', 'users/me/nodes/', { 'related_counts' : true }),
+        ];
+        self.breadcrumbs = m.prop([
+            new Breadcrumb('All My Projects','http://localhost:8000/v2/users/me/nodes/?related_counts=true', 'collection')
+        ]);
+        self.nameFilters = [
+            new Filter('Caner Uguz', '8q36f')
+        ];
+        self.tagFilters = [
+            new Filter('Something Else', 'something-else', 'tag')
+        ];
+        self.filesData = m.prop($osf.apiV2Url(
+            self.collections[0].path,
+            { query : self.collections[0].pathQuery }
+        ));
+
+        self.updateFilesData = function(linkObject) {
+            if (linkObject.link !== self.filesData()) {
+                self.filesData(linkObject.link);
+                self.isLoadedUrl = false; // check if in fact changed
+                self.updateBreadcrumbs(linkObject);
+            }
+        };
+
+        // INFORMATION PANEL
+        self.selected = m.prop([]);
+        self.updateSelected = function(selectedList){
+            self.selected(selectedList);
+        };
+
+        // COLLECTIONS PANEL
+        self.activeCollection = m.prop(1);
+        self.updateCollection = function(coll) {
+            self.activeCollection(coll.id);
+            var linkObject = new LinkObject( 'collection', coll, coll.label);
+            self.updateFilesData(linkObject);
+        };
+
+
+        // USER FILTER
+        self.activeUser = m.prop(1);
+        self.updateUserFilter = function(user) {
+            self.activeUser(user.id);
+            var linkObject = new LinkObject('user', user.data, user.label);
+            self.updateFilesData(linkObject);
+        };
+
+        // Refresh the Grid
+        self.updateList = function(element, isInit, context){
+            if(!self.isLoadedUrl) {
+                var el = element || $(self.wrapperSelector).find('.fb-main').get(0);
+                m.mount(el,
```

Comment:
Can you explain why it's necessary to call `mount` here rather than using `m.component` in the view?